app/models/activity.rb

```diff
@@ -0,0 +1,62 @@
+# Activity model inherited from public_activity gem is used to customize activities' management
+class Activity < PublicActivity::Activity
+  acts_as_readable on: :created_at
+  after_create :send_email
+
+  # The formats of notification that are supported
+  SUPPORTED_FORMATS = [:feed, :email, :popup]
+
+  # Class methods used to create an activity
+  class << self
+    # Create an activity record and store the notification formats of it
+    #
+    # @param from the one triggers this activity
```

Comment:
hmm... It may not be a user actually. For example, if a user gain an achievement, `from` will be the course. Because it is the course who gives off the achievement.